c7n/query.py

```diff
@@ -806,8 +846,36 @@ class TypeInfo(metaclass=TypeMeta):
 
     **Arn handling / generation metadata - Optional**
 
-    :param arn: Arn resource attribute, when describe format has arn
-    :param arn_type: Type, used for arn construction, also required for universal tag augment
+    :param arn: Defines a field in the resource definition that contains the ARN value, when
+        the resource has an ARM..
+
+        This value is accessed used by the 'get_arns(..)' fn on the super-class
+        QueryResourceManager. This value must be a simple field name and cannot be a path.
```

Comment:
Addressed in more depth later, but it seems totally reasonable to have this represent a JMESPath expression, and then optimize away unnecessary `jmespath.search()` calls as we like within custodian's derived `jmespath_search()` method.

Reply:
Thanks - that helps